### ISSUE TITLE
Allow implicit string concatenation for parenthesized strings

### ISSUE THREAD

**Issue by jnooree** [microsoft/pyright#2781]
**Is your feature request related to a problem? Please describe.**
Implicit string concatenation is sometimes useful, especially for long strings. Currently, the pyright checker always diagnoses it if `reportImplicitStringConcatenation` is set to true.

**Describe the solution you'd like**
Allow implicit string concatenation for parenthesized, consecutive strings. This is somewhat related to the "using assignment as a condition" warning behavior in many languages, which will be turned off once parenthesized.

**Additional context**
```python
# This will be diagnosed as usual:
a_func_with_two_args("first argument" "second argument")
#                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#                    Implicit string concatenation not allowed

# This also will:
a_func_with_single_long_arg("This is the first argument, which contains "
#                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                            "especially long text that could not fit into "
#                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                            "one single line thus should be spread.")
#                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#                           Implicit string concatenation not allowed

# While this would not:
a_func_with_single_long_arg(("This is the first argument, which contains "
                             "especially long text that could not fit into "
                             "one single line thus should be spread."))
```

**Comment by erictraut** [comment]
That sounds reasonable. This will be implemented in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.203, which I just published. It will also be included in the next release of pylance.

**Comment by JelleZijlstra** [comment]
For what it's worth, here's an example where implicit concat within parentheses is wrong: https://github.com/python/mypy/pull/11961/commits/37957b16d52d05611b4797a638939cd6f3325dd8.

**Comment by npip99** [comment]
@erictraut This issue still appears to exist in the newest version of pyright. I'm not sure if it was a regression?

**Comment by erictraut** [comment]
@npip99, this issue was fixed years ago. Here's a code sample that demonstrates the bug (as reported above) no longer exists:

```python
# pyright: reportImplicitStringConcatenation=true

def test(x: str): ...

test(("concatentated " "string"))
```

If you think that you are seeing a new bug, please open a new issue and provide a minimal, self-contained example that demonstrates the problem you're seeing.
